compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1339,7 +1339,18 @@ class RefChecks extends MiniPhase { thisPhase =>
   }
 
   override def transformTypeTree(tree: TypeTree)(using Context): TypeTree = {
-    checkExperimental(tree.symbol, tree.srcPos)
+    object CheckExperimental extends TypeTraverser {
```

Comment:
Is there any way to get more precise positions of errors with with approach? Now if an experimental type is found inside another type (which could be e.g. a complex match type spanning across multiple lines) the entire type expression will be marked as an error instead of just the reference to the experimental type

Reply:
It seems that at this phase we only have a `TypeTree` for those types. We should see if it is possible to keep them around. But maybe later.